src/net/network_manager.cpp

```diff
@@ -63,6 +63,10 @@ void ServerNetworkManager::register_listeners() {
                 go->set_type(proto::GameObject::Type::GameObject_Type_PLAYER);
             else if (dynamic_cast<Goal*>(obj))
                 go->set_type(proto::GameObject::Type::GameObject_Type_GOAL);
+            else if (dynamic_cast<Chest*>(obj))
+                go->set_type(proto::GameObject::Type::GameObject_Type_CHEST);
+            else if (dynamic_cast<Spikes*>(obj))
+                go->set_type(proto::GameObject::Type::GameObject_Type_SPIKE);
```

Comment:
This part's getting a little hairy... no action needed now, but we should think about how to simplify this in the future